org.uqbar.project.wollok.launch/src/org/uqbar/project/wollok/launch/WollokLauncherInterpreterEvaluator.xtend

```diff
@@ -42,25 +45,30 @@ class WollokLauncherInterpreterEvaluator extends WollokInterpreterEvaluator {
 	
 	override dispatch evaluate(WTest test) {
 		try {
-			wollokTestsReporter.testStart(test)
+			//wollokTestsReporter.testStart(test)
 			val x = test.elements.evalAll
 			wollokTestsReporter.reportTestOk(test)
 			x
 		}
 		catch (WollokInterpreterException e) {
 			if (e.cause instanceof AssertionException) {
-				wollokTestsReporter.reportTestAssertError(test, e.cause as AssertionException, e.lineNumber, e.objectURI)
+				wollokTestsReporter.reportTestAssertError(test, e.cause as AssertionException, null, e.lineNumber, e.objectURI)
 				null
-			}
-			else {
+			} else {
 				wollokTestsReporter.reportTestError(test, e, e.lineNumber, e.objectURI)
 				null
 			}
 		}
 		catch (WollokProgramExceptionWrapper e) {
-			// an uncaught wollok-level exception wrapped into java
-			wollokTestsReporter.reportTestError(test, e, e.lineNumber, e.URI)
-			null
+			val className = e.wollokException.call("className").toString
+			if (className.equalsIgnoreCase(ASSERTION_EXCEPTION)) {
+				wollokTestsReporter.reportTestAssertError(test, new AssertionException(e.wollokMessage), e.wollokException, e.lineNumber, e.URI)
+				null
+			} else {
+				// an uncaught wollok-level exception wrapped into java
+				wollokTestsReporter.reportTestError(test, e, e.lineNumber, e.URI)
+				null
+			}
```

Comment:
Cool, this differentiates an assertion from another type of exception.

In terms of coding I think that we could encapsulate some annoying boilerplate code into WollokProgramExceptionWrapper. Since from java I believe we will be doing this often.
We could have something like
```
if (e.isAssertionException)
```
fort this particular case.
And to check for other name

```
if (e.isWollokException("BlahException"))
```

Even just a shortcut for the name could help for later expresing something similtar to a pattern matching

```
switch (e.wollokClassName) {
    case ASSERTION_EXCEPTION: ...
    default:  ...
}
```

What do you think ?